src/PowerFSM.cpp

```diff
@@ -249,7 +254,7 @@ void PowerFSM_setup()
     // Handle press events - note: we ignore button presses when in API mode
     powerFSM.add_transition(&stateLS, &stateON, EVENT_PRESS, NULL, "Press");
     powerFSM.add_transition(&stateNB, &stateON, EVENT_PRESS, NULL, "Press");
-    powerFSM.add_transition(&stateDARK, &stateON, EVENT_PRESS, NULL, "Press");
+    powerFSM.add_transition(&stateDARK, isPowered() ? &statePOWER : &stateON, EVENT_PRESS, NULL, "Press");
```

Comment:
not sure if this is evaluated every time it is called or only once at startup. If the latter is the case, we need to move the conditional.

Reply:
The transition is only processed when event and "from state" match, in this case a press button event, which happens very rarely. The code behind isPowered() looks simple enough to be optimized by the compiler to a few assembly instructions. I don't see here much gain for any kind of optimisations. I guess there are many other low-hanging fruits at other places in the code.